crates/typst/src/eval/call.rs

```diff
@@ -32,135 +34,50 @@ impl Eval for ast::FuncCall<'_> {
         }
 
         // Try to evaluate as a call to an associated function or field.
-        let (callee, mut args) = if let ast::Expr::FieldAccess(access) = callee {
-            let target = access.target();
-            let target_span = target.span();
+        let (callee, args) = if let ast::Expr::FieldAccess(access) = callee {
+            let target_expr = access.target();
             let field = access.field();
-            let field_span = field.span();
+            let mut args = args.eval(vm)?.spanned(span);
 
+            // Evaluate the target expression.
             let target = if is_mutating_method(&field) {
-                let mut args = args.eval(vm)?.spanned(span);
-                let target = target.access(vm)?;
-
-                // Only arrays and dictionaries have mutable methods.
-                if matches!(target, Value::Array(_) | Value::Dict(_)) {
-                    args.span = span;
-                    let point = || Tracepoint::Call(Some(field.get().clone()));
-                    return call_method_mut(target, &field, args, span).trace(
-                        vm.world(),
-                        point,
-                        span,
-                    );
+                match target_expr.access(vm)? {
+                    // Only arrays and dictionaries have mutable methods.
+                    Value::Array(_) | Value::Dict(_) => {
+                        let point = || Tracepoint::Call(Some(field.get().clone()));
+                        return call_method_mut(
+                            target_expr.access(vm)?,
+                            &field,
+                            args,
+                            span,
+                        )
+                        .trace(vm.world(), point, span);
+                    }
+                    target => target.clone(),
                 }
```

Comment:
We should perhaps not access twice, if possible:

```suggestion
                match target_expr.access(vm)? {
                    // Only arrays and dictionaries have mutable methods.
                    target@(Value::Array(_) | Value::Dict(_)) => {
                        let point = || Tracepoint::Call(Some(field.get().clone()));
                        return call_method_mut(
                            target,
                            &field,
                            args,
                            span,
                        )
                        .trace(vm.world(), point, span);
                    }
                    target => target.clone(),
                }
```